### PR TITLE
[G7T5-48] Create Job Table Schema + APIs

### PR DESCRIPTION
**Pull request by charmjj**
# Description

Created table for Job + APIs for CRUD operations

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by deadline2359 — APPROVED** [review]

File: backend/app/api/api_v1/endpoints/job.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+            detail="Job not found",
+        )
+    return job
+
+@router.post("", response_model=schemas.Job)
+def create_job(
+    *,
+    db: Session = Depends(deps.get_db),
+    job_name: str = Body(...),
+    job_desc: str = Body(...),
```
Comment (jonathantan1425) [inline backend/app/api/api_v1/endpoints/job.py:49]: In the alembic SQL, `job_desc` does not have a `NOT NULL` constraint. In this case, using `Body(...)` would enforce the field to be a compulsory field in the body. If it's not compulsory, it should be `Body(None)` instead

File: backend/app/api/api_v1/endpoints/job.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    is_active: bool
```
Comment (jonathantan1425) [inline backend/app/api/api_v1/endpoints/job.py:50]: Does the lack of `= Body(None)` here work?

File: backend/app/api/api_v1/endpoints/job.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    """
+    if not settings.USERS_OPEN_REGISTRATION:
+        raise HTTPException(
+            status_code=403,
+            detail="Open staff registration is forbidden on this server",
+        )
+    job_in = schemas.JobCreate(
+        job_name=job_name,
+        job_desc=job_desc,
+        is_active=is_active,
```
Comment (jonathantan1425) [inline backend/app/api/api_v1/endpoints/job.py:63]: This implies that all three attributes are compulsory for this endpoint. Is this intentional? Also, do we want to "automate" the `is_active` value? i.e. if creating a job, `is_active` is automatically set as `True` (just a suggestion)

File: backend/app/api/api_v1/endpoints/job.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    return job
+
+@router.put("/{job_id}", response_model=schemas.Job)
+def update_job_by_id(
+    *,
+    db: Session = Depends(deps.get_db),
+    job_id: int,
+    job_name: str = Body(...),
+    job_desc: str = Body(...),
+    is_active: bool
```
Comment (jonathantan1425) [inline backend/app/api/api_v1/endpoints/job.py:75]: Same as above, this implies that every `PUT` request must have a valid set of `job_name`, `job_desc` and `is_active`

File: backend/app/api/api_v1/endpoints/role.py [hunk 5]
```diff
@@ -19,11 +19,6 @@ def get_all_role(
     Retrieve all roles.
     """
     roles = crud.role.get_multi(db, skip=skip)
-    if not roles:
-        raise HTTPException(
-            status_code=404,
-            detail="Roles not found",
-        )
```
Comment (jonathantan1425) [inline backend/app/api/api_v1/endpoints/role.py:26]: Is there a reason for removing this? Not sure if it's a commit issue

File: backend/app/api/api_v1/endpoints/role.py [hunk 6]
```diff
@@ -36,18 +31,14 @@ def get_role_by_id(
     Get a specific role by id.
     """
     role = crud.role.get(db, role_id=role_id)
-    if not role:
-        raise HTTPException(
-            status_code=404,
-            detail="Role not found",
-        )
```
Comment (jonathantan1425) [inline backend/app/api/api_v1/endpoints/role.py:43]: Same as above, could be a commit issue?

**Review by jonathantan1425 — COMMENTED** [review]
Largely should work, just a bunch of nits. Also, can you add in the unit tests?

File: backend/app/api/api_v1/endpoints/job.py (hunk 1 above)
Comment (charmjj) [reply backend/app/api/api_v1/endpoints/job.py]: changed!

File: backend/app/api/api_v1/endpoints/job.py (hunk 3 above)
Comment (charmjj) [reply backend/app/api/api_v1/endpoints/job.py]: done, now creating job only needs name and desc as is_active will be set to True by default

File: backend/app/api/api_v1/endpoints/job.py [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+    Delete a job.
+    """
+    job = crud.job.get(db, job_id=job_id)
+    if not job:
+        raise HTTPException(
+            status_code=404,
+            detail="Job not found",
+        )
+    job = crud.job.remove(db, job_id=job_id)
+    return job
```
[the remark this answers is not in the document]
Comment (charmjj) [reply backend/app/api/api_v1/endpoints/job.py]: done, oops

File: backend/app/api/api_v1/endpoints/job.py (hunk 2 above)
Comment (charmjj) [reply backend/app/api/api_v1/endpoints/job.py]: updated to not require is_active during job creation

File: backend/app/api/api_v1/endpoints/job.py (hunk 4 above)
Comment (charmjj) [reply backend/app/api/api_v1/endpoints/job.py]: done

File: backend/app/models/job.py [hunk 8]
```diff
@@ -0,0 +1,10 @@
+from email.policy import default
+from sqlalchemy import Column, Integer, String, Boolean
+
+from app.db.base_class import Base
+
+class Job(Base):
+    job_id = Column(Integer, primary_key=True, index=True)
+    job_name = Column(String, nullable=False)
+    job_desc = Column(String)
+    is_active = Column(Boolean, nullable=False, default=True)
```
[the remark this answers is not in the document]
Comment (charmjj) [reply backend/app/models/job.py]: done

File: backend/app/api/api_v1/endpoints/role.py (hunk 6 above)
Comment (charmjj) [reply backend/app/api/api_v1/endpoints/role.py]: think it's a commit issue, i dont recall committing this file at all. Fixed, sry about that!

File: backend/app/api/api_v1/endpoints/role.py [hunk 9]
```diff
@@ -36,18 +31,14 @@ def get_role_by_id(
...
     return role
 
 
 @router.post("", response_model=schemas.Role)
 def create_role(
+    *,
     db: Session = Depends(deps.get_db),
-    role_name: str = Body(None, embed=True),
+    role_name: str = Body(...),
```
[the remark this answers is not in the document]
Comment (charmjj) [reply backend/app/api/api_v1/endpoints/role.py]: fixed

File: backend/app/api/api_v1/endpoints/role.py [hunk 10]
```diff
@@ -69,7 +60,7 @@ def update_role_by_id(
     *,
     db: Session = Depends(deps.get_db),
     role_id: int,
-    role_name: str = Body(None, embed=True),
+    role_name: str = Body(None),
```
[the remark this answers is not in the document]
Comment (charmjj) [reply backend/app/api/api_v1/endpoints/role.py]: fixed

File: backend/app/api/api_v1/endpoints/role.py (hunk 5 above)
Comment (charmjj) [reply backend/app/api/api_v1/endpoints/role.py]: think it's a commit issue, ive added it back in!

**Comment by quinncheong** [comment]
Lint via black before committing

File: backend/app/tests/crud/test_job.py [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+#     item_in = ItemCreate(title=title, description=description)
+#     user = create_random_user(db)
+#     item = crud.item.create_with_owner(db=db, obj_in=item_in, owner_id=user.id)
+#     item2 = crud.item.remove(db=db, id=item.id)
+#     item3 = crud.item.get(db=db, id=item.id)
+#     assert item3 is None
+#     assert item2.id == item.id
+#     assert item2.title == title
+#     assert item2.description == description
+#     assert item2.owner_id == user.id
```
Comment (jonathantan1425) [inline backend/app/tests/crud/test_job.py:70]: Can remove

**Review by jonathantan1425 — APPROVED** [review]

**Pull request merged**